JavaScript/JavaScriptSDK/AppInsights.ts

```diff
@@ -250,11 +257,18 @@ module Microsoft.ApplicationInsights {
         }
 
         public trackAjax(absoluteUrl: string, pathName: string, totalTime: number, success: boolean, resultCode: number) {
-            var dependency = new Telemetry.RemoteDependencyData(absoluteUrl, pathName, totalTime, success, resultCode);
-            var dependencyData = new ApplicationInsights.Telemetry.Common.Data<ApplicationInsights.Telemetry.RemoteDependencyData>(
-                Telemetry.RemoteDependencyData.dataType, dependency);
-            var envelope = new Telemetry.Common.Envelope(dependencyData, "Microsoft.ApplicationInsights." + this.config.instrumentationKey.replace(/-/g, "") + ".RemoteDependency");
-            this.context.track(envelope);
+            if (this.config.maxAjaxCallsPerView === -1 ||
+                this._ajaxCallsSent < this.config.maxAjaxCallsPerView) {
+                var dependency = new Telemetry.RemoteDependencyData(absoluteUrl, pathName, totalTime, success, resultCode);
+                var dependencyData = new ApplicationInsights.Telemetry.Common.Data<ApplicationInsights.Telemetry.RemoteDependencyData>(
+                    Telemetry.RemoteDependencyData.dataType, dependency);
+                var envelope = new Telemetry.Common.Envelope(dependencyData, "Microsoft.ApplicationInsights." + this.config.instrumentationKey.replace(/-/g, "") + ".RemoteDependency");
+                this.context.track(envelope);
+                ++this._ajaxCallsSent;
+            } else {
+                _InternalLogging.throwInternalUserActionable(LoggingSeverity.CRITICAL,
```

Comment:
Include page name to the message. Also including pageView.id would be great if we can somehow show this trace message to be related to that pageview. Maybe even show it on the pageview details blade.

Also - would it make sense to only send it once per pageview? Otherwise we send as many traces as ajaxes were on the page.